src/assets/scripts/client/aircraft/FlightManagementSystem/LegModel.js

```diff
@@ -1,538 +1,780 @@
+        this.init(navigationLibrary, routeString);
     }
 
     /**
-     * Return the name of the procedure if this is a procedure leg
+     * Returns the active `WaypointModel`
      *
-     * @property procedureName
-     * @type {string}
-     * @return {string}
+     * Assumed to always be the first item in the `#waypointCollection`
+     *
+     * @for LegModel
+     * @property currentWaypoint
+     * @type {WaypointModel}
      */
-    get procedureName() {
-        if (!this.isProcedure) {
-            return null;
+    get currentWaypoint() {
+        if (this._waypointCollection.length === 0) {
+            throw new TypeError('Expected the current leg to contain at least one waypoint');
         }
 
-        return this._procedureRouteModel.procedure;
+        return this._waypointCollection[0];
     }
 
     /**
-     * Return the name of the procedure exit (airport) if this is a procedure leg
+     * Returns whether this leg is an airway leg
      *
-     * This is used only for arrival routes and only within the `AircraftStripView`
+     * @for LegModel
+     * @property isAirwayLeg
+     * @type {boolean}
+     */
+    get isAirwayLeg() {
+        return this._legType === LEG_TYPE.AIRWAY;
+    }
+
+    /**
+     * Returns whether this leg is a direct leg
      *
-     * @property exitName
-     * @type {string}
-     * @return {string}
+     * @for LegModel
+     * @property isDirectLeg
+     * @type {boolean}
      */
-    get exitName() {
-        if (!this.isProcedure) {
-            return null;
-        }
+    get isDirectLeg() {
+        return this._legType === LEG_TYPE.DIRECT;
+    }
 
-        return this._procedureRouteModel.exit;
+    /**
+     * Returns whether this leg is a procedure leg
+     *
+     * @for LegModel
+     * @property isProcedureLeg
+     * @type {boolean}
+     */
+    get isProcedureLeg() {
+        return this._legType === LEG_TYPE.PROCEDURE;
     }
 
     /**
-     * Return the name and exit of the procedure if this is a procedure leg
+     * Whether this leg is a SID Procedure leg
      *
-     * Will return a string in an abbreviated procedureRouteString format.
+     * @for RouteModel
+     * @property isSidLeg
+     * @type {boolean}
+     */
+    get isSidLeg() {
+        return this.isProcedureLeg && this._procedureModel.procedureType === PROCEDURE_TYPE.SID;
+    }
+
+    /**
+     * Whether this leg is a STAR Procedure leg
      *
-     * This should only be used for the view, like the `AircraftStripView`, because
-     * the output will not be valid for anything that works with a routeString
+     * @for RouteModel
+     * @property isStarLeg
+     * @type {boolean}
+     */
+    get isStarLeg() {
+        return this.isProcedureLeg && this._procedureModel.procedureType === PROCEDURE_TYPE.STAR;
+    }
+
+    /**
+     * Returns the type of this leg
      *
-     * @property procedureAndExitName
+     * @for LegModel
+     * @property legType
      * @type {string}
```

Comment:
is this a `string` or a property of an enum?  If the latter, that should be the type instead of `string`.